File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/room/EtkRoomDB.java
package com.faciee.cti.valbastrelu.eticket.room;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

import com.faciee.cti.valbastrelu.eticket.room.dao.BiletDao;
import com.faciee.cti.valbastrelu.eticket.room.dao.BiletPDao;
import com.faciee.cti.valbastrelu.eticket.room.dao.StatieDao;
import com.faciee.cti.valbastrelu.eticket.room.dao.TranzactieDao;
import com.faciee.cti.valbastrelu.eticket.room.entities.Bilet;
import com.faciee.cti.valbastrelu.eticket.room.entities.BiletP;
import com.faciee.cti.valbastrelu.eticket.room.entities.Statie;
import com.faciee.cti.valbastrelu.eticket.room.entities.Tranzactie;

@Database(entities = {Bilet.class, BiletP.class, Tranzactie.class, Statie.class}, version = 6, exportSchema = false)
public abstract class EtkRoomDB extends RoomDatabase {
	
	public abstract BiletDao biletDao();
	public abstract BiletPDao biletpDao();
	public abstract TranzactieDao tranzactieDao();
	public abstract StatieDao statieDao();
	private static EtkRoomDB INSTANCE;
	
	public static EtkRoomDB getDatabase(final Context context) {
		if (INSTANCE == null) {
			synchronized (EtkRoomDB.class) {
				if (INSTANCE == null) {
					INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
							EtkRoomDB.class, "word_database")
							// Wipes and rebuilds instead of migrating if no Migration object.
							// Migration is not part of this codelab.
							.fallbackToDestructiveMigration()
							.build();
				}
			}
		}
		return INSTANCE;
	}
}

File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/room/EtkRoomDB.kt
package com.faciee.cti.valbastrelu.eticket.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.faciee.cti.valbastrelu.eticket.base.SingletonHolder
import com.faciee.cti.valbastrelu.eticket.room.converter.DateConverter
import com.faciee.cti.valbastrelu.eticket.room.converter.PriceConverter
import com.faciee.cti.valbastrelu.eticket.room.converter.TransactionTypeConverter
import com.faciee.cti.valbastrelu.eticket.room.converter.TransportTypeConverter
import com.faciee.cti.valbastrelu.eticket.room.dao.RoutesDao
import com.faciee.cti.valbastrelu.eticket.room.dao.StationDao
import com.faciee.cti.valbastrelu.eticket.room.dao.TicketDao
import com.faciee.cti.valbastrelu.eticket.room.dao.TicketParkingDao
import com.faciee.cti.valbastrelu.eticket.room.dao.TransactionsDao
import com.faciee.cti.valbastrelu.eticket.room.entities.Route
import com.faciee.cti.valbastrelu.eticket.room.entities.Station
import com.faciee.cti.valbastrelu.eticket.room.entities.Ticket
import com.faciee.cti.valbastrelu.eticket.room.entities.TicketParking
import com.faciee.cti.valbastrelu.eticket.room.entities.Transaction

@Database(entities = [Ticket::class, TicketParking::class, Transaction::class, Station::class, Route::class], version = 1, exportSchema = false)
@TypeConverters(DateConverter::class, TransportTypeConverter::class, PriceConverter::class, TransactionTypeConverter::class)
abstract class EtkRoomDB : RoomDatabase() {
	abstract fun ticketDao(): TicketDao
	abstract fun ticketParkingDao(): TicketParkingDao
	abstract fun transactionsDao(): TransactionsDao
	abstract fun stationDao(): StationDao
	abstract fun routesDao(): RoutesDao

	companion object : SingletonHolder<EtkRoomDB, Context>({
		Room.databaseBuilder(it.applicationContext,
				EtkRoomDB::class.java, "word_database")
				.fallbackToDestructiveMigration()
				.build()
	})
}